File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/20.go
package main
import (
    "fmt"
    "strings"
)

func exatamenteUmAb(s string) bool {
    return strings.Count(s, "ab") == 1
}

func main() {
    fmt.Println(exatamenteUmAb("abab"))  // False
    fmt.Println(exatamenteUmAb("aab"))   // True
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/20.java
public class AFDExatamenteUmAb {
    public static boolean exatamenteUmAb(String s) {
        int countAb = s.split("ab", -1).length - 1;
        return countAb == 1;
    }

    public static void main(String[] args) {
        System.out.println(exatamenteUmAb("abab"));  // False
        System.out.println(exatamenteUmAb("aab"));   // True
    }
}
